appGeo/src/tests/game_rules.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { GameRules, SUCCESS, FAILURE } from '~/app/game/game-rules.component';
+
+QUnit.test("Constructor and Setting", function(assert) {
+    const gameRule = new GameRules();
+
+    assert.true(gameRule.isScheduledEnd(), "Scheduled End is true by default")
+    assert.true(gameRule.isWipeoutEnd(), "Wipeout End is true by default")
+    assert.false(gameRule.isTestingOverrule(), "Testing Overrule is true by default")
+
+    assert.equal(gameRule.getMinPlayers(), 5, "Default min players is 5")
+    assert.equal(gameRule.getFractionKillers(), 0.2, "Default fraction killers is 0.2")
+    assert.equal(gameRule.getGameLengthHours(), 168, "Default game length is one week")
+    assert.equal(gameRule.getDayCycleLength(), 1440, "Default game cycle is 24 hours")
+    assert.equal(gameRule.getSafeLength(), 480, "Default safe period is 8 hours")
+    assert.equal(gameRule.getVoteLength(), 60, "Default vote period is 1 hour")
+    assert.equal(gameRule.getVoteTime, 960, "Default vote time is 16 hours into game")
+    assert.equal(gameRule.getMaxSoloKill, 1, "Default maximum kills per day by a killer is 1")
+    assert.equal(gameRule.getMaxGlobalKill, Infinity, "Default maximum kills per day overall is infinity")
```

Comment:
here you are defining variable such as gameRule.getVoteTime; getMaxSoloKill; and getMaxGlobalKill. Shouldn't these be function calls and should be gameRule.getVoteTime(); getMaxSoloKill(); and getMaxGlobalKill()?